TwilioVerifySDKTests/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorageTests.swift

```diff
@@ -0,0 +1,242 @@
+//
+//  AuthenticatedSecureStorageTests.swift
+//  TwilioVerifySDKTests
+//
+//  Copyright © 2021 Twilio.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+import LocalAuthentication
+@testable import TwilioVerifySDK
+
+class AuthenticatedSecureStorageTests: XCTestCase {
+
+  var testSubject: AuthenticatedSecureStorage!
+  var keychainMock: KeychainMock!
+
+  private enum MockErrors: Error {
+    case unexpected
+  }
+
+  override func setUp() {
+    keychainMock = .init()
+    testSubject = .init(keychain: keychainMock, keychainQuery: KeychainQuery())
+  }
+
+  // MARK: - Save Data
+
+  func testSave_withValidValuesSavingData_shouldNotThrow() {
+    // Given
+    let data = "testData".data(using: .utf8)
+    let key = "testKey"
+    let authenticator = AuthenticatorMock(context: MockContext())
+    keychainMock.addItemStatus = [errSecSuccess]
+
+    // When
+    testSubject.save(data!, withKey: key, authenticator: authenticator) { [self] in
+      // Then
+      XCTAssertEqual(self.keychainMock.callsToAddItem, 1)
```

Comment:
Asking because I don't know what rules or guidelines we use for testing: should we add the error in case this Assert doesn't work? e.g `XCTAssertEqual(self.keychainMock.callsToAddItem, 1, "unable to validate calls to add item")`

Reply:
No defined yet, but if you ask me, I prefer to add it only if is needed. IE: When is quite hard to understand the comparison/expectation.